robyn/openapi.py

```diff
@@ -0,0 +1,267 @@
+            openapi_object["components"]["responses"] = component_responses
+
+        if component_parameters:
+            openapi_object["components"]["parameters"] = component_parameters
+
+        if component_examples:
+            openapi_object["components"]["examples"] = component_examples
+
+        if component_request_bodies:
+            openapi_object["components"]["requestBodies"] = component_request_bodies
+
+        if component_security_schemes:
+            openapi_object["components"]["securitySchemes"] = component_security_schemes
+
+        if component_links:
+            openapi_object["components"]["links"] = component_links
+
+        if component_callbacks:
+            openapi_object["components"]["callbacks"] = component_callbacks
+
+        if component_path_items:
+            openapi_object["components"]["pathItems"] = component_path_items
+
+        return openapi_object
+
+    def add_openapi_path_obj(self, route_type: str, endpoint: str, openapi_summary: str, openapi_tags: list, signature: Signature = None):
+        """
+        adds the given path to openapi spec
+
+        :param route_type: the http method as string (get, post ...)
+        :param endpoint: the endpoint to be added
+        :param openapi_summary: short summary of the endpoint (to be fetched from the endpoint defenition by default)
+        :param openapi_tags: tags -- for grouping of endpoints
+        :param signature: the function signature -- to grab the typed dict annotations for query params
+        """
+
+        query_params = None
+
+        if "query_params" in signature.parameters:
+            query_params = signature.parameters["query_params"].default
+
+        modified_endpoint, path_obj = self.get_path_obj(endpoint, openapi_summary, openapi_tags, query_params)
+
+        if modified_endpoint not in self.openapi_schema["paths"]:
+            self.openapi_schema["paths"][modified_endpoint] = {}
+        self.openapi_schema["paths"][modified_endpoint][route_type] = path_obj
+
+    def add_subrouter_paths(self, subrouter_openapi):
+        """
+
+        adds the subrouter paths to main router's openapi specs
+
+        :param subrouter_openapi: the OpenAPI object of the current subrouter
+        """
+        paths = subrouter_openapi.openapi_schema["paths"]
+
+        for path in paths:
+            self.openapi_schema["paths"][path] = paths[path]
+
+    def get_path_obj(self, endpoint: str, summary: str, tags: list, query_params: TypedDict = None):
+        """
+        get the "path" openapi object according to spec
+
+        :param endpoint: the endpoint to be added
+        :param summary: short summary of the endpoint (to be fetched from the endpoint defenition by default)
+        :param tags: tags -- for grouping of endpoints
+        :param query_params: query params for the function
+
+        :return: the "path" openapi object according to spec
+        """
+        modified_endpoint = endpoint
```

Comment:
how is it modified here?

Reply:
Robyn uses `:param` for path params & swagger expects `{param}` -- this is being done here.